src/read-package-json.ts

```diff
@@ -1,10 +1,12 @@
-import * as fs from 'fs-extra';
-import * as path from 'path';
+import fs from 'graceful-fs';
+import path from 'node:path';
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 export async function readPackageJson(dir: string, safe = false): Promise<any> {
   try {
-    return await fs.readJson(path.resolve(dir, 'package.json'));
+    return JSON.parse(await fs.promises.readFile(path.resolve(dir, 'package.json'), {
```

Comment:
Since this is a `graceful-fs` call, it'll need to be passed through `util.promisify`